--- a/app/app.py
+++ b/app/app.py
@@ -4,14 +4,13 @@
 import os
 
 app = Flask(__name__)
-# Database configuration (replace with your credentials)
-db_config = {
-    "user": "beyblade_user",
-    "password": "Sample_DB_Password",
-    "host": "db",
-    "database": "beyblade_db"
-}
+
+# Database configuration (using environment variables)
+app.config['MYSQL_HOST'] = os.environ.get('DB_HOST', 'db')
+app.config['MYSQL_USER'] = os.environ.get('DB_USER', 'beyblade_user')
+app.config['MYSQL_PASSWORD'] = os.environ.get('DB_PASSWORD', 'Sample_DB_Password')
+app.config['MYSQL_DB'] = os.environ.get('DB_NAME', 'beyblade_db')
+
 
 
 def get_db_connection():
@@ -19,10 +18,7 @@
         conn = mysql.connector.connect(
             host=app.config['MYSQL_HOST'],
             user=app.config['MYSQL_USER'],
-            password=db_config["password"],
+            password=app.config['MYSQL_PASSWORD'],
             database=app.config['MYSQL_DB']
-
-
-            database=db_config["database"]
         )
         return conn
     except mysql.connector.Error as err:
@@ -34,11 +30,12 @@
     if conn is None:
         return None
     cursor = conn.cursor()
-    cursor.execute(f"SELECT id FROM {table} WHERE name = %s", (name,))
-    result = cursor.fetchone()
-    if conn:
-        conn.close() 
+    try:
+      cursor.execute(f"SELECT id FROM {table} WHERE name = %s", (name,))
+      result = cursor.fetchone()
+      return result[0] if result else None
+    except mysql.connector.Error as e:
+      print(f"Error executing query: {e}")
     return result[0] if result else None